### ISSUE TITLE
Translator implicitly converts null to (infinite) Interval[null, null]

### ISSUE THREAD

**Issue by cmoesel** [cqframework/clinical_quality_language#596]
Given the following expression:
```
define NullBeforeInterval:
  (null as Integer) before Interval[1, 10]
```
the translator does an implicit conversion of `null` to an `Interval<Integer>` in order to match the signature for `Before`.  But, it produces it with closed boundaries:
```JSON
{
   "lowClosed" : true,
   "highClosed" : true,
   "type" : "Interval",
   "low" : {
      "strict" : false,
      "type" : "As",
      "operand" : {
         "type" : "Null"
      },
      "asTypeSpecifier" : {
         "name" : "{urn:hl7-org:elm-types:r1}Integer",
         "type" : "NamedTypeSpecifier"
      }
   },
   "high" : {
      "strict" : false,
      "type" : "As",
      "operand" : {
         "type" : "Null"
      },
      "asTypeSpecifier" : {
         "name" : "{urn:hl7-org:elm-types:r1}Integer",
         "type" : "NamedTypeSpecifier"
      }
   }
}
```

This is a problem because a null closed boundary is considered the min/max of the type range, so the translator has translated `null` as `Interval[null, null]`, which is effectively `Interval[-2147483648, 2147483647]` -- and of course ends up w/ unintended semantics.  In the CQL example above, it causes `(null as Integer) before Interval[1, 10]` to be evaluated as `false` rather than the correct answer, `null`.

**Comment by cmoesel** [comment]
Of course this exposes a potential runtime problem with converting points to intervals as well.  If the translator blindly constructs a closed interval w/ the same point on both ends, then this works fine as long as the point doesn't evaluate to null -- but if the point evaluates to null, it really ought to use open boundaries.  So translation of point X to Interval Y should probably be something more like:
```
if X is null then Interval(X, X) else Interval[X, X]
```

**Comment by cmoesel** [comment]
If you're going to fix this for 1.5, could you also backport it to 1.4?  That's actually where I found the problem (but then confirmed it was also the same in 1.5).

**Comment by brynrhodes** [comment]
I actually think the correct translation here is:

```
if X is null then null else Interval[X, X]
```

Any concerns with that @cmoesel ?

**Comment by cmoesel** [comment]
No problem with that.  I think I was assuming you needed an `Interval` shape to keep the translator happy.  My only question is if it should be just be plain `null` then or `null as Interval<Integer>` to maintain the intended signature?

**Comment by brynrhodes** [comment]
It will be a typed null, but the ELM Null has that capability. In CQL you'd have to do an explicit cast, but since I'm already building out ELM, I can just type the Null as I construct it.

**Comment by cmoesel** [comment]
Sounds good to me!

**Comment by brynrhodes** [comment]
Addressed in the 1.5.2 release
